### PR TITLE
Acoustic comms : Packet collision timeout

### PR DESCRIPTION
**Pull request by adityapande-1995**
🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎉 New feature

## Summary
This PR adds a new tag ``<collision_time_interval>`` to the acoustic comms plugin. If an address endpoint receives two data packets within that time interval, the second one will be dropped. The timeout value defaults to zero, so no packets are ever dropped if the tag is unspecified.

The time must be in seconds.

## Test it
Added an integration test using ``acoustic_comms_packet_collision.sdf``. If multiple packets are sent during a 1 second time interval, they will be dropped. In the test, I send 3 packets, and only the first one of them is received.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

**Review by hidmic — COMMENTED** [review]
LGTM, but I don't follow what phenomena we are trying to model here.

File: src/systems/acoustic_comms/AcousticComms.cc [hunk 1]
```diff
@@ -175,8 +190,41 @@ void AcousticComms::Step(
...
+              // This is the first message received by this address.
+              receivedSuccessfully = true;
+            }
+            else
+            {
+              // A previous msg was already received at this address.
+              std::chrono::duration<double> timeGap = currTimestamp -
+                this->dataPtr->lastMsgReceivedTime[msg->dst_address()];
+
+              auto dropInterval = std::chrono::duration<double>(
```
Comment (arjo129) [inline src/systems/acoustic_comms/AcousticComms.cc:209]: Knit: From a logical stand point wouldn't the dropped packet also clog the reciever for a fixed amount of time.

**Review by arjo129 — COMMENTED** [review]
Seems reasonable to me. However, I would encourage us to actually allow users to set a "bit-rate" so that the packet collision also depends on the length of the packet. I don't believe this is too much extra work but would lead to a much more realistic 
and usable model.

File: src/systems/acoustic_comms/AcousticComms.cc (hunk 1 above)
Comment (adityapande-1995) [reply src/systems/acoustic_comms/AcousticComms.cc:216]: Done https://github.com/gazebosim/gz-sim/pull/1755/commits/25b975ae3b6d7c0f48ad3a428bf1a951179b01dc

**Review by hidmic — APPROVED** [review]

**Comment by adityapande-1995** [comment]
I'm pretty sure this is not breaking ABI, but the ABI checker is failing weirdly.

**Comment by mjcarroll** [comment]
> I'm pretty sure this is not breaking ABI, but the ABI checker is failing weirdly.

ABI checker is having some issues at the moment.

**Pull request merged**